extensions/amp-pan-zoom/0.1/amp-pan-zoom.js

```diff
@@ -285,23 +279,48 @@ export class AmpPanZoom extends AMP.BaseElement {
   }
 
   /**
-   * @return {number}
-   * @private
+   * Calculate the width and height of the content dimensions such
+   * that they fit within amp-pan-zoom.
+   * @param {number} aspectRatio
    */
-  getYOffsetFromCenter_() {
-    const elementContentGap = this.elementBox_.height - this.contentBox_.height;
-    const distanceToCenter = elementContentGap / 2;
-    return this.contentBox_.top - distanceToCenter;
+  calculateContentDimensions_(aspectRatio) {
+    // Calculate content height if we set width to amp-pan-zoom's width
+    const heightToFit = this.elementBox_.width / aspectRatio;
+    // Calculate content width if we set height to be amp-pan-zoom's height
+    const widthToFit = this.elementBox_.height * aspectRatio;
+
+    // The content should fit within amp-pan-zoom, so take the smaller value
+    let height = Math.min(heightToFit, this.elementBox_.height);
+    let width = Math.min(widthToFit, this.elementBox_.width);
+
+    if (Math.abs(width - this.sourceWidth_) <= 16
+        && Math.abs(height - this.sourceHeight_) <= 16) {
+      width = this.sourceWidth_;
+      height = this.sourceHeight_;
+    }
+
+    this.contentBox_ = layoutRectLtwh(
+        0,
+        0,
+        Math.round(width),
+        Math.round(height));
   }
 
   /**
-   * @return {number}
-   * @private
+   * Adjust max scale to at least fit the screen. This handles
+   * the case of ridiculously long or wide content. We guarantee
+   * that when zoomed to max, the smaller dimension fits the entire
+   * amp-pan-zoom space.
+   * @param {number} sourceAspectRatio
    */
-  getXOffsetFromCenter_() {
-    const elementContentGap = this.elementBox_.width - this.contentBox_.width;
-    const distanceToCenter = elementContentGap / 2;
-    return this.contentBox_.left - distanceToCenter;
+  calculateMaxScale_(sourceAspectRatio) {
```

Comment:
since this has a side-effect, might be better named `updateMaxScale`

Reply:
Updated this for `calculateContentDimensions` -> `updateContentDimensions` as well for the same reason.